app/src/main/java/com/android/voyageur/ui/trip/activities/ActivitiesMapScreen.kt

```diff
@@ -0,0 +1,275 @@
+package com.android.voyageur.ui.trip.activities
+
+import android.app.DatePickerDialog
+import android.widget.DatePicker
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.FilterAlt
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.android.voyageur.R
+import com.android.voyageur.model.activity.Activity
+import com.android.voyageur.model.activity.ActivityType
+import com.android.voyageur.model.trip.TripsViewModel
+import com.android.voyageur.ui.components.SearchBar
+import com.google.android.gms.maps.model.LatLng
+import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.Marker
+import com.google.maps.android.compose.MarkerState
+import com.google.maps.android.compose.rememberCameraPositionState
+import java.util.*
+
+// Default location for Lausanne, Switzerland
+val LAUSANNE_COORDINATES = LatLng(46.519962, 6.633597)
+
+/**
+ * A screen that displays a map of activities for the selected trip. Users can search for activities
+ * and filter them by date and type.
+ *
+ * @param tripsViewModel The `TripsViewModel` used to manage trip-related data and logic.
+ * @param onActivitiesChanged callback only used for testing to notify when activities have changed.
+ *   As we cannot put testTags on google map markers
+ */
+@Composable
+fun ActivitiesMapTab(
+    tripsViewModel: TripsViewModel,
+    onActivitiesChanged: (List<Activity>) -> Unit = {}
+) {
+  var activities by remember { mutableStateOf(tripsViewModel.getActivitiesForSelectedTrip()) }
+  val cameraPositionState = rememberCameraPositionState {
```

Comment:
I think that if the trip has already a location maybe we should put the default location in that place, rather than Lausanne. Otherwise, Lausanne is fine if the trip doesn't have a chosen location. Let me know your thoughts.

Reply:
That's exactly what happens if there are no activities with locations it defaults to Lausanne :)

Reply:
Hence the null check before closing Lausanne default

Reply:
Ah yes, you're right, my mistake.